### ISSUE TITLE
[YCQL] CREATE INDEX doesn't honor keyspace.table_name

### ISSUE THREAD

**Issue by ajcaldera1** [yugabyte/yugabyte-db#4253]
If I attempt to `CREATE INDEX` like the following:
```
CREATE INDEX mykeyspace.foo_bar_idx ON mykeyspace.foo_bar (col_name);
```
I get the following:
```
Invalid SQL statement. syntax error, unexpected '.', expecting ON
```

**Comment by kmuthukk** [comment]
@ajcaldera1 -- This might be a behavior cloned from Apache Cassandra.

Instead of this:
`CREATE INDEX mykeyspace.foo_bar_idx ON mykeyspace.foo_bar (col_name);`
can you try:
`CREATE INDEX foo_bar_idx ON mykeyspace.foo_bar (col_name);`
?

We might be syntactically restricting the user from specifying a keyspace for the index, and the index is expected to be in the same keyspace as the table. I suppose one could relax the syntax restriction (for readability) and only have a semantic restriction that the two keyspaces match.

**Comment by m-iancu** [comment]
@ajcaldera1 @kmuthukk Indeed, the index keyspace is implied by the (indexed) table's keyspace so it is at best redundant. Therefore, the syntax expects the index name to be a simple name rather than a qualified name.

**Comment by ajcaldera1** [comment]
It would be nice to do one of the following:
1) Return a nicer message like "Keyspace qualification is not supported"
-or-
2) Compare the qualification to the table_name and if it matches, simply proceed as if no qualification had been passed (ignore it) and if it doesn't match, return "Keyspace for index cannot be different from table"
